lte/gateway/python/integ_tests/s1aptests/s1ap_utils.py

```diff
@@ -1115,11 +1126,32 @@ def disable_service(self, service):
         Args:
             service: (str) service to disable
         """
+        service_name = self.get_service_name_from_init_system(service)
         if self._init_system == InitMode.SYSTEMD:
-            self.exec_command(f"sudo systemctl mask magma@{service}")
-            self.exec_command(f"sudo systemctl stop magma@{service}")
+            self.exec_command(f"sudo systemctl mask {service_name}")
+            self.exec_command(f"sudo systemctl stop {service_name}")
         elif self._init_system == InitMode.DOCKER:
-            self.exec_command(f"docker stop {service}")
+            self.exec_command(f"docker stop {service_name}")
+
+    def check_if_magma_services_are_active(self) -> bool:
+        """check if all services in the list are active (only works for docker
+         init_system)
+
+        Returns:
+            (bool) True if all services are active, False otherwise
+        """
+        magma_services = {
+            "mme", "magmad", "sctpd", "sessiond", "policydb", "state",
+            "directoryd", "connectiond", "td-agent-bit", "redis",
+            "subscriberdb", "eventd", "mobilityd", "pipelined", "monitord",
+            "envoy_controller", "smsd", "enodebd", "redirectd", "ctraced",
+            "control_proxy",
+        }
+        for service in magma_services:
+            if not self.is_service_active(service):
+                print(f"************* Waiting for {service} to restart")
```

Comment:
This function is supposed to check if the services are active, but is not supposed to know whether the caller function is waiting or not for the service to restart. The print can be modified to something:
```
print(f"************* {service} is not running")
```